### PR TITLE
Fixed the URL Encoding used in the logs and the prefix comparison

### PR DESCRIPTION
**Pull request by apluche**
Fixed the encoding to properly match URIs containing spaces in both the prefix matching and in the logs.

### CODE REVIEW

File: CredentialProvider.Microsoft.Tests/Logging/LoggingTests.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public void HumanFriendlyTextWriterLogger_EmitsLogLevelAndMessage()
             mockWriter.Setup(x => x.WriteLine(It.IsAny<string>()));
             HumanFriendlyTextWriterLogger logger = new HumanFriendlyTextWriterLogger(mockWriter.Object, writesToConsole: false);
             logger.SetLogLevel(LogLevel.Error);
-            logger.Log(LogLevel.Error, allowOnConsole: true, "Something bad happened");
+            logger.Log(LogLevel.Error, true, "Something bad happened");
```
Comment (jmyersmsft) [inline CredentialProvider.Microsoft.Tests/Logging/LoggingTests.cs:20]: please put the argument name back here

File: CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs [hunk 2]
```diff
@@ -115,7 +115,7 @@ public override async Task<GetAuthenticationCredentialsResponse> HandleRequestAs
 
                     if (!string.IsNullOrWhiteSpace(sessionToken))
                     {
-                        Verbose(string.Format(Resources.VSTSSessionTokenCreated, request.Uri.ToString()));
+                        Verbose(string.Format(Resources.VSTSSessionTokenCreated, request.Uri.AbsoluteUri.ToString()));
```
Comment (jmyersmsft) [inline CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs:118]: `AbsoluteUri` is already a `string` (this occurs in several places)

Comment (apluche) [reply CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs]: done

File: CredentialProvider.Microsoft.Tests/Logging/LoggingTests.cs (hunk 1 above)
Comment (apluche) [reply CredentialProvider.Microsoft.Tests/Logging/LoggingTests.cs]: done

**Comment by shadargee1982** [comment]
@apluche - only question I have is around tests.  Does it make sense to add 1-2 unit tests?

**Comment by apluche** [comment]
@shadargee1982 not in this case since the only these changes do is make sure that the Uri string is properly encoded when put into the logs or doing a string comparison.

Ex. 
old code: "Test Project"
new code: "Test%20Project"

**Pull request merged**